hoomd/mpcd/ReverseNonequilibriumShearFlow.cc

```diff
@@ -0,0 +1,276 @@
+ * \param slab_distance Slab distance
+ * \param p_target target momentum in x-direction for the two slabs
+ */
+mpcd::ReverseNonequilibriumShearFlow::ReverseNonequilibriumShearFlow(
+    std::shared_ptr<SystemDefinition> sysdef,
+    std::shared_ptr<Trigger> trigger,
+    unsigned int num_swap,
+    Scalar slab_width,
+    Scalar slab_distance,
+    Scalar p_target)
+    : Updater(sysdef, trigger), m_mpcd_pdata(sysdef->getMPCDParticleData()), m_num_swap(num_swap),
+      m_slab_width(slab_width), m_slab_distance(slab_distance), m_momentum_exchange(0), m_num_lo(0),
+      m_num_hi(0), m_p_target(p_target), m_update_slabs(true)
+    {
+    m_exec_conf->msg->notice(5) << "Constructing ReverseNonequilibriumShearFlow" << std::endl;
+
+    // subscribe to box change signal, to ensure geometry is still valid
+    m_pdata->getBoxChangeSignal()
+        .connect<mpcd::ReverseNonequilibriumShearFlow,
+                 &mpcd::ReverseNonequilibriumShearFlow::requestUpdateSlabs>(this);
+    // allocate memory
+    GPUArray<Scalar2> lo_idx(m_num_swap, m_exec_conf);
+    m_layer_lo.swap(lo_idx);
+    GPUArray<Scalar2> hi_idx(m_num_swap, m_exec_conf);
+    m_layer_hi.swap(hi_idx);
+    }
+
+mpcd::ReverseNonequilibriumShearFlow::~ReverseNonequilibriumShearFlow()
+    {
+    m_exec_conf->msg->notice(5) << "Destroying ReverseNonequilibriumShearFlow" << std::endl;
+    m_pdata->getBoxChangeSignal()
+        .disconnect<mpcd::ReverseNonequilibriumShearFlow,
+                    &mpcd::ReverseNonequilibriumShearFlow::requestUpdateSlabs>(this);
+    }
+
+/*!
+ * \param new_num_swap Max number of swaps
+ */
+void mpcd::ReverseNonequilibriumShearFlow::setNswap(unsigned int new_num_swap)
+    {
+    if (new_num_swap > m_num_swap)
+        {
+        // allocate memory if needed
+        GPUArray<Scalar2> lo_idx(new_num_swap, m_exec_conf);
+        m_layer_lo.swap(lo_idx);
+        GPUArray<Scalar2> hi_idx(new_num_swap, m_exec_conf);
+        m_layer_hi.swap(hi_idx);
+        }
+    m_num_swap = new_num_swap;
+    }
+
+//! Set the target velocity
+void mpcd::ReverseNonequilibriumShearFlow::setTargetMomentum(Scalar p_target)
+    {
+    m_p_target = p_target;
+    }
+
+/*!
+ * \param slab_width Slab thickness
+ */
+void mpcd::ReverseNonequilibriumShearFlow::setSlabWidth(Scalar slab_width)
+    {
+    // check that delta is bigger than zero
+    if (slab_width <= Scalar(0))
+        {
+        throw std::runtime_error("invalid slab thickness in ReverseNonequilibriumShearFlow");
+        }
+
+    m_slab_width = slab_width;
+    requestUpdateSlabs();
+    }
+
+/*!
+ * \param slab_separation Slab separation
+ */
+void mpcd::ReverseNonequilibriumShearFlow::setSlabSeparation(Scalar slab_separation)
+    {
+    // check that delta is bigger than zero
+    if (slab_separation <= Scalar(0))
+        {
+        throw std::runtime_error("invalid slab separation in ReverseNonequilibriumShearFlow");
+        }
+
+    m_slab_distance = Scalar(0.5) * slab_separation;
```

Comment:
I checked and `m_slab_distance` is not used in very many places, so I suggest we this rename the variable to `m_slab_separation`, remove this scaling, and then multiply `m_slab_separation` by a half when it's needed in `setSlabs` like we do with `m_slab_width`.

Reply:
@mphoward I went ahead to resolve these